Clarify CTS monitoring of service and instances (#12008)
hashicorp/consul: website/content/docs/nia/index.mdx

-> **Note:** The terminology "tasks" used throughout the documentation refers to all types of tasks except when specifically stated otherwise.

**Source Input** - A source input defines objects that provide values or metadata to the Terraform module. See [source input](/docs/nia/terraform-modules#source-input) for the supported metadata and values. For example, a user can configure a Consul KV source input to provide KV pairs as variables to their respective Terraform module. The source input can be included in two ways. It can be specified as a parameter in a condition using `source_includes_var` and also by using the `source_input` block.

**Network Infrastructure Automation (NIA)** - Enables dynamic updates to network infrastructure devices triggered when specific conditions, such as service changes and registration, are met.

> **Services** - A service in CTS represents a service that is registered with Consul for service discovery. Services are grouped by its service name that may have one or more instances of that service with unique ids. CTS monitors services based on service names, and can provide service instance details to a Terraform module for network automation.

Reviewer: It's not completely clear on what you're trying to say in the second sentence (`Services are grouped by . . .`).

maybe:
```suggestion
**Services** - A service in CTS represents a service that is registered with Consul for service discovery. Services are grouped by their service names, as there may be more than one instances of a particular service, each with their own unique ID. CTS monitors services based on service names, and can provide service instance details to a Terraform module for network automation.
```

Reviewer: Another possible rework:

`**Services** - A service in CTS represents a service that is registered with Consul for service discovery. Since more than one instance of a service can exist, each with their own unique id, CTS monitors services based on service names. CTS monitors these services and provides service instance details to a Terraform module for network automation.`

Author: Sweet, I like your suggestion and it makes what I was trying to convey more clearly.

> **Services** - A service in CTS represents a service that is registered with Consul for service discovery. Services are grouped by its service name that may have one or more instances of that service with unique ids. CTS monitors services based on service names, and can provide service instance details to a Terraform module for network automation.

Reviewer 2: The `its` is throwing me off a little, I think because you have `Services`. Maybe `Services are grouped by a service name`?

Author: thanks for calling out `its`, I like @wilkermichael's suggestion to replace with `their`

> Services are grouped by their service names

> **Services** - A service in CTS represents a service that is registered with Consul for service discovery. Services are grouped by their service names, as there may be more than one instances of a particular service, each with their own unique ID. CTS monitors services based on service names, and can provide service instance details to a Terraform module for network automation.

Reviewer 3: ```suggestion
**Services** - A service in CTS represents a service that is registered with Consul for service discovery. Services are grouped by their service names. There may be more than one instance of a particular service, each with its own unique ID. CTS monitors services based on service names and can provide service instance details to a Terraform module for network automation.
```
I believe 'its' must agree with 'each' in terms of plurality.

Author: Thanks for the review, adding it in 👍
